src/Microsoft.DocAsCode.Build.TableOfContents/TocResolver.cs

```diff
@@ -59,56 +105,91 @@ private TocItemInfo ResolveItemCore(TocItemInfo wrapper, Stack<FileAndType> stac
+                            Logger.LogWarning($"Href {item.Href} is overwritten by tocHref {item.TocHref}");
+                            var tocFilePath = (RelativePath)file.File + (RelativePath)item.TocHref;
+
+                            var tocFile = new FileAndType(file.BaseDir, tocFilePath, file.Type, file.PathRewriter);
+
                             if (!_collection.TryGetValue(tocFile, out tocFileModel))
                             {
-                                var message = $"Unable to find either {Constants.YamlTocFileName} or {Constants.MarkdownTocFileName} inside {item.Href}. Make sure the file is included in config file docfx.json!";
+                                var message =
+                                    $"Unable to find {item.TocHref}. Make sure the file is included in config file docfx.json!";
                                 Logger.LogWarning(message);
                                 break;
                             }
                         }
+                        else
+                        {
+                            var relativeFolder = (RelativePath)file.File + (RelativePath)item.Href;
+                            var tocFilePath = relativeFolder + (RelativePath)Constants.YamlTocFileName;
+
+                            var tocFile = new FileAndType(file.BaseDir, tocFilePath, file.Type, file.PathRewriter);
+
+                            // First, try finding toc.yml under the relative folder
+                            // Second, try finding toc.md under the relative folder
+                            if (!_collection.TryGetValue(tocFile, out tocFileModel))
+                            {
+                                tocFilePath = relativeFolder + (RelativePath)Constants.MarkdownTocFileName;
+                                tocFile = new FileAndType(file.BaseDir, tocFilePath, file.Type, file.PathRewriter);
+                                if (!_collection.TryGetValue(tocFile, out tocFileModel))
+                                {
+                                    var message =
+                                        $"Unable to find either {Constants.YamlTocFileName} or {Constants.MarkdownTocFileName} inside {item.Href}. Make sure the file is included in config file docfx.json!";
+                                    Logger.LogWarning(message);
+                                    break;
+                                }
+                            }
 
-                        item.TocHref = tocFilePath - (RelativePath)file.File;
+                            item.TocHref = tocFilePath - (RelativePath)file.File;
+                        }
 
                         // Get homepage from the referenced toc
-                        if (string.IsNullOrEmpty(item.Homepage) && string.IsNullOrEmpty(item.HomepageUid))
+                        if (string.IsNullOrEmpty(item.TopicHref) && string.IsNullOrEmpty(item.TopicUid))
```

Comment:
Yes... Add this logic when href/topicHref/topicUid is all null.

---
In reply to: [73801641](https://github.com/dotnet/docfx/pull/537#discussion_r73801641) [](ancestors = 73801641)